Mathlib/Topology/Separation.lean

```diff
@@ -2100,6 +2166,36 @@ lemma SeparatedNhds.of_isCompact_isClosed {s t : Set X}
 @[deprecated (since := "2024-01-28")]
 alias separatedNhds_of_isCompact_isClosed := SeparatedNhds.of_isCompact_isClosed
 
+/-- This technique to produce a SeparatingCover in regular Lindelöf topological spaces
+will be used to prove regular Lindelöf spaces are normal. -/
+lemma SeparatingCover.of_isClosed {s t : Set X} [r: RegularSpace X] [LindelofSpace X]
+    (s_cl : IsClosed s) (t_cl : IsClosed t) (st_dis : Disjoint s t) : SeparatingCover s t := by
+  -- `IsLindelof.indexed_countable_subcover` requires the space be Nonempty
+  wlog nonempty_X : Nonempty X
+  · have : s = ∅ := subset_eq_empty (fun ⦃_⦄ _ ↦ trivial)
+      (univ_eq_empty_iff.mpr (not_nonempty_iff.mp nonempty_X))
+    rw [this]
+    apply And.left
```

Comment:
This a really weird move that is bound to lead to a confusing tactic state since Lean has no way to guess what would be the right hand-side. Can’t you simply stick a `|>.1` at the end of the next line? Of course using parentheses is also fine.

Reply:
210a9f0